### PR TITLE
Restful branch

### PR DESCRIPTION
**Pull request by JACornelius**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,12 +8,26 @@
     <groupId>twitterapp.src</groupId>
     <artifactId>TwitterApp</artifactId>
     <version>1.0-SNAPSHOT</version>
+    <properties>
+        <dropwizard.version>
```
Comment (kev-chan) [inline pom.xml]: please actually use the properties.  You probably want to update the dropwizard version here too

File: src/main/java/twitterapp/src/Tweeting.java [hunk 2]
```diff
@@ -1 +1 @@
...
            try {
                twitter.updateStatus(message);
            } catch (TwitterException te) {
                te.printStackTrace();
            }
        }

    }
}
/**/
```
Comment (kev-chan) [inline src/main/java/twitterapp/src/Tweeting.java]: the formatting for this file is odd.  Can you fix this?

File: src/main/java/twitterapp/src/TwitterAppResource.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package twitterapp.src;
+
+import com.codahale.metrics.annotation.Timed;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.glassfish.jersey.server.spi.ResponseErrorMapper;
+import twitter4j.*;
```
Comment (kev-chan) [inline src/main/java/twitterapp/src/TwitterAppResource.java]: let's not use *s for importing.  It's an intellij setting

File: src/main/java/twitterapp/src/TwitterAppResource.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            //String tweet = "trying to tweet with dropwizard round 2";
+            //t.updateStatus(tweet);
+            //newStatusUpdate = newStatusUpdate + tweet;
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+        }
+        System.out.println("Code 200: Timeline has been printed.");
+        return Response.ok(timeline, MediaType.TEXT_PLAIN).build();
```
Comment (kev-chan) [inline src/main/java/twitterapp/src/TwitterAppResource.java]: return the JSON representation of the information and not a string

File: src/main/java/twitterapp/src/TwitterAppResource.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        Twitter t = TwitterFactory.getSingleton();
+        try{
+            List<Status> statuses = t.getHomeTimeline();
+
+            for(Status s: statuses)
+            {
+                timeline = timeline + s.getUser().getName() + ": " + s.getText() + "\n";
+            }
+
+            //String tweet = "trying to tweet with dropwizard round 2";
```
Comment (kev-chan) [inline src/main/java/twitterapp/src/TwitterAppResource.java]: remove commented out code if it isn't needed

File: src/main/java/twitterapp/src/TwitterAppResource.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    else
+    {
+        try {
+            t.updateStatus(tweet);
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+        System.out.print("Code 200: Tweet has been posted.");
+        return Response.status(200).entity("Tweet has been posted").build();
```
Comment (kev-chan) [inline src/main/java/twitterapp/src/TwitterAppResource.java]: what will happen if t.updateStatus fails?

File: src/main/java/twitterapp/src/TwitterAppResource.java (hunk 4 above)
Comment (kev-chan) [reply src/main/java/twitterapp/src/TwitterAppResource.java]: what happens if the getHomeTimeline call fails?

**Review by kev-chan — CHANGES_REQUESTED** [review]

**Comment by JACornelius** [comment]
I have fixed all the changes but the weird formatting on tweeting.java. Not so sure what you mean by weird formattig

**Comment by JACornelius** [comment]
Nevermind I got the weird formatting fixed.

File: TwitterApp.iml [hunk 7]
```diff
@@ -1,14 +1,16 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<module type="JAVA_MODULE" version="4">
-  <component name="NewModuleRootManager">
-    <output url="file://$MODULE_DIR$/out/production/TwitterApp" />
-    <output-test url="file://$MODULE_DIR$/out/test/TwitterApp" />
-    <exclude-output />
+<module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule="true" type="JAVA_MODULE" version="4">
```
Comment (kev-chan) [inline TwitterApp.iml]: let's remove this file

File: pom.xml [hunk 8]
```diff
@@ -8,12 +8,25 @@
...
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+    </properties>
+
     <dependencies>
         <dependency>
-            <groupId>org.twitter4j</groupId>
-            <artifactId>twitter4j-core</artifactId>
-            <version>4.0.4</version>
+            <groupId>io.dropwizard</groupId>
```
Comment (kev-chan) [inline pom.xml:21]: nice job here

File: src/main/java/twitterapp/src/TwitterAppResource.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+        System.out.println("Code 500: Tweet is too long, keep it within in 280 characters");
+       return Response.status(500).entity("Tweet is too long, keep it with in 280 characters").build();
+    }
+    else {
+        try {
+            t.updateStatus(tweet);
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+            System.out.println("Code 500: Tweet was not posted, possibly due to authentication error or duplicated tweet");
```
Comment (kev-chan) [inline src/main/java/twitterapp/src/TwitterAppResource.java]: please respond with an actual 500 error code in the response

File: src/main/java/twitterapp/src/TwitterAppResource.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+            timeline = "Printing home timeline: \n";
+            for(Status s: statuses)
+            {
+                timeline = timeline + s.getUser().getName() + ": " + s.getText() + "\n";
+            }
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+            System.out.println("Code 500: There was a problem, possibly incorrect authentication codes");
```
Comment (kev-chan) [inline src/main/java/twitterapp/src/TwitterAppResource.java]: please respond with an actual 500 error code in the response

File: src/main/java/twitterapp/src/TwitterAppResource.java [hunk 11]
```diff
@@ -0,0 +1,68 @@
+package twitterapp.src;
```
Comment (kev-chan) [inline src/main/java/twitterapp/src/TwitterAppResource.java:1]: please test your error cases

**Review by kev-chan — CHANGES_REQUESTED** [review]

**Comment by JACornelius** [comment]
it should be all good with handling error cases.

File: src/main/java/twitterapp/src/TwitterAppResource.java [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+    static final int MAX_LENGTH = 280;
+
+    @GET
+    @Path("/timeline")
+    public Response getTimeline()
+    {
+        String timeline ="";
+        Twitter t = TwitterFactory.getSingleton();
+        try{
+            List<Status> statuses = t.getHomeTimeline();
```
Comment (kev-chan) [inline src/main/java/twitterapp/src/TwitterAppResource.java]: please return back all of the status info in json

**Review by kev-chan — CHANGES_REQUESTED** [review]

**Comment by kev-chan** [comment]
update the error message for authentcation error scenarios

**Pull request closed without merging**